android/src/main/java/com/swmansion/rnscreens/ScreenStack.java

```diff
@@ -231,22 +220,27 @@ protected void performUpdate() {
       }
     }
 
+    ArrayList<ScreenStackFragment> visibleScreensBelowTop = getVisibleScreensBelowTop();
+
     for (ScreenStackFragment screen : mScreenFragments) {
       // detach all screens that should not be visible
-      if (screen != newTop && screen != belowTop && !mDismissed.contains(screen)) {
+      if (screen != newTop && !visibleScreensBelowTop.contains(screen) && !mDismissed.contains(screen)) {
         getOrCreateTransaction().remove(screen);
       }
     }
-    // attach "below top" screen if set
-    if (belowTop != null && !belowTop.isAdded()) {
-      final ScreenStackFragment top = newTop;
-      getOrCreateTransaction().add(getId(), belowTop).runOnCommit(new Runnable() {
-        @Override
-        public void run() {
-          top.getScreen().bringToFront();
-        }
-      });
+    // attach screens visible below top
+    for (int i = visibleScreensBelowTop.size() - 1; i >= 0; i--){
```

Comment:
Can we keep the previous simple loops in place instead of allocating copies of screens array so many times here? I'm fine with extracing it to a method with a more descriptive name. Note that it is a bit unintuitive why all the arrays are in the stack order while visibleScreensBelowTop array is reverse